### ISSUE TITLE
Feature request: running without MQTT server

### ISSUE THREAD

**Issue by mitkodotcom** [arendst/Sonoff-MQTT-OTA-Arduino#176]
If there is no MQTT server running on the configured host, the application stops responding for about 10 seconds every 30 second or so. It hangs in mqtt_reconnect() until mqttClient.connect times out. Meanwhile the web server does not accept new connections which may brake IFTTT or similar scripts. 

I made a (very) quick and dirty fix by adding:

if (strlen(sysCfg.mqtt_host) == 0) { mqttcounter = 60; return; }

at the beginning of mqtt_reconnect()

BTW, is it possible to disable MQTT at compile time?

**Comment by arendst** [comment]
MQTT is the core of this software.

**Comment by mitkodotcom** [comment]
Yes, and that's great, but it is more convenient to use one firmware for all purposes.  It's quite easy and harmless to add an "if there is no server configured, do not try to connect" statement.

Meanwhile I found another more elegant solution to my "hangs for 15 seconds every 30 seconds" problem. I have just decreased MQTT_SOCKET_TIMEOUT from 15 to 2. It should not take 15 seconds for the server to respond anyway.
